### ISSUE TITLE
[FORGE] Incompatibility with Create 0.5.1b

### ISSUE THREAD

**Issue by Siarko** [ChiselsAndBits/Chisels-and-Bits#1123]
Hello
Minecraft version: 1.19.2
C&B version: 1.3.135
Create version: 0.5.1.b

During dedicated server startup I'm getting error:

```[25Jun2023 19:52:57.556] [modloading-worker-0/ERROR] [mod.chiselsandbits.api.util.ClassUtils/]: Failed to load: mod.chiselsandbits.forge.compat.create.CreateCandBPlugin
java.lang.NoClassDefFoundError: com/simibubi/create/content/contraptions/components/structureMovement/MovementBehaviour
        at java.lang.Class.forName0(Native Method) ~[?:?]
        at java.lang.Class.forName(Unknown Source) ~[?:?]
        at mod.chiselsandbits.api.util.ClassUtils.createOrGetInstance(ClassUtils.java:45) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at mod.chiselsandbits.forge.platform.ForgePluginDiscoverer.createPluginFrom(ForgePluginDiscoverer.java:95) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at mod.chiselsandbits.forge.platform.ForgePluginDiscoverer.loadPlugins(ForgePluginDiscoverer.java:58) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at mod.chiselsandbits.plugin.PluginManger.detect(PluginManger.java:35) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at mod.chiselsandbits.ChiselsAndBits.<init>(ChiselsAndBits.java:62) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at mod.chiselsandbits.forge.Forge.lambda$new$1(Forge.java:42) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at com.communi.suggestu.scena.core.init.PlatformInitializationHandler.onInit(PlatformInitializationHandler.java:29) ~[scena-forge-1.0.95.jar%23332!/:1.0.95]
        at mod.chiselsandbits.forge.Forge.<init>(Forge.java:37) ~[chisels-and-bits-forge-1.3.135.jar%23218!/:1.3.135]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
        at java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:?]
        at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:68) ~[javafmllanguage-1.19.2-43.2.11.jar%23311!/:?]
        at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:121) ~[fmlcore-1.19.2-43.2.11.jar%23310!/:?]
        at java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source) [?:?]
        at java.util.concurrent.CompletableFuture$AsyncRun.exec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool.scan(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) [?:?]
Caused by: java.lang.ClassNotFoundException: com.simibubi.create.content.contraptions.components.structureMovement.MovementBehaviour
        at jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source) ~[?:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
        at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.4.jar:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
        at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.4.jar:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
        ... 24 more
```

Server is starting correctly but custom blocks are not attached to contraptions and don't move with them.
Looks like there were some changes in class structure of create between 0.5.0 and 0.5.1. 
Are there plans to support create 0.5.1b for minecraft 1.19.2?

**Comment by The-Arcanian** [comment]
Happens with Create 0.5.1c as well.

**Comment by mobbymobslayer** [comment]
its because create trains cant move immovable objects (blocks that cant be moved by pistons or movable contraption)  meaning the mod cant move them, ive managed to determine its because in versions of create past 0.5.0i they've added physics mechanics that make blocks like obsidian and armoured deepslate unmovable for create machinery, however it looks like they've programmed it for any block that's coded as unbreakable, and there's only configs to change the intractability with obsidian armoured deepslate and mob spanners, it might be worth asking to add an additional config setting for "other immovable blocks" witch would fix this issue with chisel and bits compatibility,

**Comment by KubLuk24** [comment]
Having the same issue here, but on Fabric with the newest Create mod. I'm trying to either port Chisels and Bits in a way that the blocks are not registered as immovable or edit create so that it skips these checks.

Also, to anyone thinking about trying it with Carpet's _movableBlockEntities_ system, it won't work.

**Comment by vgrynch** [comment]
I wonder, can we just unmark Chisels and Bits blocks to be movable? Would it cause much of the issues?

**Comment by marchermans** [comment]
Yes that would break but I just merged a pr for 1.19.2 a couple of days ago which fixes it
